File: fix_encoding.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import re

# Read the file
with open(r'd:\Antigravity\OBSOLIO-FE\src\translations\index.js', 'r', encoding='utf-8') as f:
    lines = f.readlines()

# Fix line 783 (index 782) - the Arabic startFreeTrial
# We'll replace any non-ASCII characters in that specific line with the correct Arabic
for i, line in enumerate(lines):
    if i == 782:  # Line 783 (0-indexed)
        if 'startFreeTrial' in line:
            lines[i] = "        startFreeTrial: 'جرّب مجاناً',\r\n"
            print(f"Fixed line {i+1}: startFreeTrial")
    
# Write back
with open(r'd:\Antigravity\OBSOLIO-FE\src\translations\index.js', 'w', encoding='utf-8', newline='') as f:
    f.writelines(lines)

print("Done!")

File: fix-encoding.js
const fs = require('fs');

// Read the file
const filePath = 'd:\\Antigravity\\OBSOLIO-FE\\src\\translations\\index.js';
let content = fs.readFileSync(filePath, 'utf8');

// Define replacements - these are the mojibake patterns that need to be fixed
const replacements = [
    // Register button
    ["startFreeTrial: 'Ø¬Ø±Ù'Ø¨ Ù…Ø¬Ø§Ù†Ø§Ù‹'", "startFreeTrial: 'جرّب مجاناً'"],
    // Book demo in hero section  
    ["bookDemo: 'Ø§Ø­Ø¬Ø² Ø¬Ù„Ø³Ø© Ø¹Ø±Ø¶ ØªØ¬Ø±ÙŠØ¨ÙŠ'", "bookDemo: 'احجز جلسة عرض تجريبي'"],
];

// Apply replacements
replacements.forEach(([oldText, newText]) => {
    if (content.includes(oldText)) {
        content = content.replace(oldText, newText);
        console.log(`✓ Fixed: ${oldText.substring(0, 30)}...`);
    } else {
        console.log(`✗ Not found: ${oldText.substring(0, 30)}...`);
    }
});

// Write back
fs.writeFileSync(filePath, content, 'utf8');
console.log('\nEncoding fixes applied!');
